core/src/Kokkos_CopyViews.hpp

```diff
@@ -1434,8 +1434,8 @@ inline void deep_copy(
   }
 #endif
 
+  Kokkos::DefaultHostExecutionSpace().fence();
```

Comment:
No. This needs to be `Kokkos::fence()`. There are explicit overloads of `Kokkos::deep_copy()` which do not fence everything. This one does.

Reply:
Ah, I see. I couldn't find anything about the existence of the other overloads in the documentation. We might want to add a page for `deep_copy`. I'll revert all the changes here.